update timeseries to reflect NULL filling (#15512)
apache/druid: docs/querying/timeseriesquery.md

row even if the query is run in "descending" mode. Post-aggregations in the grand totals row will be computed based
upon the grand total aggregations.

## Empty bucket values

> By default Druid fills empty interior time buckets in responses to timeseries queries with the default value for the [aggregator function](./sql-aggregations.md).

Reviewer: >in responses to
Should this be “in response to”? Or “in the results of”?

> By default Druid fills empty interior time buckets in responses to timeseries queries with the default value for the [aggregator function](./sql-aggregations.md).

Author: ```suggestion
By default Druid fills empty interior time buckets in the results of timeseries queries with the default value for the [aggregator function](./sql-aggregations.md).
```

row even if the query is run in "descending" mode. Post-aggregations in the grand totals row will be computed based
upon the grand total aggregations.

> ## NULL filling

Reviewer 2: I wonder if we should call this "Empty filling" or "Empty bucket filling" or something similar?

It doesn't actually always fill with zeros or nulls depending on the mode, rather it fills with the default values of the aggregator functions. Since they aggregate no rows, only the default value is there. https://github.com/apache/druid/blob/master/docs/querying/sql-aggregations.md has the default values of the aggregators.

> By default timeseries queries fill empty interior time buckets with NULL.

Reviewer 2: same comment about default aggregator value(s), and in other places too.

]
```

Time buckets that lie completely outside the data interval are not filled with the default value.

> You can disable all NULL filling with the context flag "skipEmptyBuckets".

Author: ```suggestion
You can disable all empty bucket filling with the context flag "skipEmptyBuckets".
```

> You can disable all empty bucket filling with the context flag "skipEmptyBuckets".

Reviewer: ```suggestion
You can disable all empty bucket filling with the context flag `skipEmptyBuckets`.
```
